qiskit/transpiler/passes/optimization/finalize_layouts.py

```diff
@@ -0,0 +1,77 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2024
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+
+"""Finalize layout-related attributes."""
+
+import logging
+
+from qiskit.transpiler.basepasses import AnalysisPass
+from qiskit.transpiler.layout import Layout
+
+logger = logging.getLogger(__name__)
+
+
+class FinalizeLayouts(AnalysisPass):
+    """Finalize 'layout' and 'final_layout' attributes, taking 'virtual_permutation_layout'
+    into account when exists.
+    """
+
+    def run(self, dag):
+        """Run the FinalizeLayouts pass on ``dag``.
+
+        Args:
+            dag (DAGCircuit): the DAG circuit.
+        """
+
+        if (
+            virtual_permutation_layout := self.property_set.get("virtual_permutation_layout", None)
+        ) is None:
```

Comment:
Not that it matters, but `property_set` is a default dict that returns `None` if the key isn't present.

```suggestion
        if (
            virtual_permutation_layout := self.property_set["virtual_permutation_layout"]
        ) is None:
```